src/System.Text.Json/tests/Imported/DateTimeConverterTests.cs

```diff
@@ -0,0 +1,176 @@
+            c.DateTimeField = localDateTime;
+            c.PreField = "Pre";
+            c.PostField = "Post";
+            string json = JsonSerializer.ToString(c); 
+            c.DateTimeField = new DateTime(2008, 1, 1, 1, 1, 1, 0, DateTimeKind.Utc);
+            string json2 = JsonSerializer.ToString(c); 
+
+            TimeSpan offset = TimeZoneInfo.Local.GetUtcOffset(localDateTime);
+
+            // if the current timezone is utc then local already equals utc
+            if (offset == TimeSpan.Zero)
+            {
+                Assert.Equal(json, json2);
+            }
+            else
+            {
+                Assert.NotEqual(json, json2);
+            }
+        }
+
+        [Fact]
+        public void DeserializeDateTimeOffset()
+        {
+            // Intentionally use an offset that is unlikely in the real world,
+            // so the test will be accurate regardless of the local time zone setting.
+            var offset = new TimeSpan(2, 15, 0);
+            var dto = new DateTimeOffset(2014, 1, 1, 0, 0, 0, 0, offset);
+
+            var test = JsonSerializer.Parse<DateTimeOffset>("\"2014-01-01T00:00:00+02:15\"");
+
+            Assert.Equal(dto, test);
+            Assert.Equal(dto.ToString("o"), test.ToString("o"));
+        }
+
+        [Fact]
+        public void NullableSerializeUTC()
+        {
+            NullableDateTimeTestClass c = new NullableDateTimeTestClass();
+            c.DateTimeField = new DateTime(2008, 12, 12, 12, 12, 12, 0, DateTimeKind.Utc).ToLocalTime();
+            c.DateTimeOffsetField = new DateTime(2008, 12, 12, 12, 12, 12, 0, DateTimeKind.Utc).ToLocalTime();
+            c.PreField = "Pre";
+            c.PostField = "Post";
+            string json = JsonSerializer.ToString(c);
+            Assert.Equal(@"{""PreField"":""Pre"",""DateTimeField"":""2008-12-12T04:12:12-08:00"",""DateTimeOffsetField"":""2008-12-12T04:12:12-08:00"",""PostField"":""Post""}", json);
+            //test the other edge case too
+            c.DateTimeField = null;
+            c.DateTimeOffsetField = null;
+            c.PreField = "Pre";
+            c.PostField = "Post";
+            json = JsonSerializer.ToString(c);
+            Assert.Equal(@"{""PreField"":""Pre"",""DateTimeField"":null,""DateTimeOffsetField"":null,""PostField"":""Post""}", json);
+        }
+
+        [Fact]
+        public void SerializeUTC()
+        {
+            DateTimeTestClass c = new DateTimeTestClass();
+            c.DateTimeField = new DateTime(2008, 12, 12, 12, 12, 12, 0, DateTimeKind.Utc).ToLocalTime();
+            c.DateTimeOffsetField = new DateTime(2008, 12, 12, 12, 12, 12, 0, DateTimeKind.Utc).ToLocalTime();
+            c.PreField = "Pre";
+            c.PostField = "Post";
+            string json = JsonSerializer.ToString(c);
+            Assert.Equal(@"{""PreField"":""Pre"",""DateTimeField"":""2008-12-12T04:12:12-08:00"",""DateTimeOffsetField"":""2008-12-12T04:12:12-08:00"",""PostField"":""Post""}", json);
```

Comment:
Some of the DateTime tests are [failing](https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F38153~2Fmerge/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190602.19/workItem/System.Text.Json.Tests/analysis/xunit/System.Text.Json.Serialization.Tests.DateTimeConverterTests~2FSerializeUTC) because of asserts like this. The json writer writes DateTime values according to the machine's local time.

`Windows.81.Amd64.Open-x64-Debug`
```
Assert.Equal() Failure
                                 ↓ (pos 46)
Expected: ···eField\":\"2008-12-12T04:12:12-08:00\",\"DateTimeOffsetField\":\"20···
Actual:   ···eField\":\"2008-12-12T12:12:12+00:00\",\"DateTimeOffsetField\":\"20···
                                 ↑ (pos 46)
```

`RedHat.6.Amd64.Open-x64-Release`

```
Assert.Equal() Failure
                                 ↓ (pos 47)
Expected: ···Field\":\"2008-12-12T04:12:12-08:00\",\"DateTimeOffsetField\":\"200···
Actual:   ···Field\":\"2008-12-12T07:12:12-05:00\",\"DateTimeOffsetField\":\"200···
                                 ↑ (pos 47)
```

I'm not entirely sure, but I think the Json.NET tests pass because of extra config (e.g. `DateTimeStyles.AssumeUniversal` used [here](https://github.com/JamesNK/Newtonsoft.Json/blob/7217c484e9705b5e76585c8b7fcd489c8e021c23/Src/Newtonsoft.Json.Tests/Converters/IsoDateTimeConverterTests.cs#L187)) that forces the UTC offset to be a particular value. We currently don't support any config.

Having these tests would require asserting against strings that account for the local time of the machine rather than this hardcoded datetime value.

Reply:
will check thanks